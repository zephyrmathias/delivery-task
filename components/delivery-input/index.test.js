import React from 'react'
import { shallow } from 'enzyme'
import DeliveryData from './index'

const props = {
  town1: '',
  town2: '',
  cost: '',
  onChangeTown1: jest.fn(),
  onChangeTown2: jest.fn(),
  onChangeCost: jest.fn(),
  onSubmitInput: jest.fn(),
}

test('should match snapshot', () => {
  const component = shallow(<DeliveryData {...props} />)
  expect(component).toMatchSnapshot()
})

test('should show an error message when it contains an error', () => {
  const component = shallow(<DeliveryData {...props} error="error" />)
  expect(component).toMatchSnapshot()
})

test('should show a list of routes when it contains the data', () => {
  const data = [{ start: 'A', end: 'B', cost: 2 }]
  const component = shallow(<DeliveryData {...props} data={data} />)
  expect(component).toMatchSnapshot()
})
