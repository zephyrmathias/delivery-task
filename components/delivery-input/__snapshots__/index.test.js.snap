// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="delivery-data__container"
>
  <Title
    title="Add Delivery Route"
  />
  <Subtitle
    subtitle="Add Delivery Data"
  />
  <Note>
    *Note: Example data: 
    <b>
      A, B, 10
    </b>
     please note
    <b>
       Town name is case sensitive
    </b>
     and 
    <b>
      Cost cannot be 0
    </b>
    <div>
      <u>
        <b>
          And Delivery Routes cannot be duplicated
        </b>
      </u>
      <div>
        Ex: if A-B-10 exist, cannot add another A-B with any cost
      </div>
    </div>
  </Note>
  <input
    className="delivery-data__input"
    name="data-town1"
    onChange={[MockFunction]}
    placeholder="Town 1"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    name="data-town2"
    onChange={[MockFunction]}
    placeholder="Town 2"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    min={1}
    name="data-cost"
    onChange={[MockFunction]}
    placeholder="Cost"
    type="number"
    value=""
  />
  <Button
    onClick={[MockFunction]}
    text="Add New Data"
  />
</div>
`;

exports[`should show a list of routes when it contains the data 1`] = `
<div
  className="delivery-data__container"
>
  <Title
    title="Add Delivery Route"
  />
  <Subtitle
    subtitle="Add Delivery Data"
  />
  <Note>
    *Note: Example data: 
    <b>
      A, B, 10
    </b>
     please note
    <b>
       Town name is case sensitive
    </b>
     and 
    <b>
      Cost cannot be 0
    </b>
    <div>
      <u>
        <b>
          And Delivery Routes cannot be duplicated
        </b>
      </u>
      <div>
        Ex: if A-B-10 exist, cannot add another A-B with any cost
      </div>
    </div>
  </Note>
  <input
    className="delivery-data__input"
    name="data-town1"
    onChange={[MockFunction]}
    placeholder="Town 1"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    name="data-town2"
    onChange={[MockFunction]}
    placeholder="Town 2"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    min={1}
    name="data-cost"
    onChange={[MockFunction]}
    placeholder="Cost"
    type="number"
    value=""
  />
  <Button
    onClick={[MockFunction]}
    text="Add New Data"
  />
  <Result>
    <div
      key="A-B"
    >
      A
      -
      B
      -
      2
    </div>
  </Result>
</div>
`;

exports[`should show an error message when it contains an error 1`] = `
<div
  className="delivery-data__container"
>
  <Title
    title="Add Delivery Route"
  />
  <Subtitle
    subtitle="Add Delivery Data"
  />
  <Note>
    *Note: Example data: 
    <b>
      A, B, 10
    </b>
     please note
    <b>
       Town name is case sensitive
    </b>
     and 
    <b>
      Cost cannot be 0
    </b>
    <div>
      <u>
        <b>
          And Delivery Routes cannot be duplicated
        </b>
      </u>
      <div>
        Ex: if A-B-10 exist, cannot add another A-B with any cost
      </div>
    </div>
  </Note>
  <input
    className="delivery-data__input"
    name="data-town1"
    onChange={[MockFunction]}
    placeholder="Town 1"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    name="data-town2"
    onChange={[MockFunction]}
    placeholder="Town 2"
    type="text"
    value=""
  />
  <input
    className="delivery-data__input"
    min={1}
    name="data-cost"
    onChange={[MockFunction]}
    placeholder="Cost"
    type="number"
    value=""
  />
  <Button
    onClick={[MockFunction]}
    text="Add New Data"
  />
  <Error>
    ERROR: 
    error
  </Error>
</div>
`;
